docs/csharp/language-reference/attributes/caller-information.md

```diff
@@ -47,19 +26,51 @@ You specify an explicit default value for each optional parameter. You can't app
 
 The following chart shows the member names that are returned when you use the `CallerMemberName` attribute.
 
-|Calls occur within|Member name result|
+| Calls occur within | Member name result |
 |-|-|
-|Method, property, or event|The name of the method, property, or event from which the call originated.|
-|Constructor|The string ".ctor"|
-|Static constructor|The string ".cctor"|
-|Finalizer|The string "Finalize"|
-|User-defined operators or conversions|The generated name for the member, for example, "op_Addition".|
-|Attribute constructor|The name of the method or property to which the attribute is applied. If the attribute is any element within a member (such as a parameter, a return value, or a generic type parameter), this result is the name of the member that's associated with that element.|
-|No containing member (for example, assembly-level or attributes that are applied to types)|The default value of the optional parameter.|
+| Method, property, or event | The name of the method, property, or event from which the call originated.|
+| Constructor | The string ".ctor" |
+| Static constructor | The string ".cctor" |
+| Finalizer | The string "Finalize" |
+| User-defined operators or conversions | The generated name for the member, for example, "op_Addition". |
+| Attribute constructor | The name of the method or property to which the attribute is applied. If the attribute is any element within a member (such as a parameter, a return value, or a generic type parameter), this result is the name of the member that's associated with that element. |
+| No containing member (for example, assembly-level or attributes that are applied to types) | The default value of the optional parameter. |
+
+## Argument expressions
+
+You use the <xref:System.Runtime.CompilerServices.CallerArgumentExpressionAttribute?displayProperty=nameWithType> when you want the expression used for an argument. Diagnostic libraries may want to provide more details about the *expressions* used for arguments. By providing the expression that triggered the diagnostic, in addition to the parameter name, developers see more details about the code that triggered the diagnostic. That extra information makes it easier to fix.
+
+The following example shows how you can provide detailed information about the argument when it's invalid:
+
+:::code language="csharp" source="./snippets/CallerInformation.cs" id="TestCondition":::
+
+You would invoke it as shown in the following example:
+
+:::code language="csharp" source="./snippets/CallerInformation.cs" id="InvokeTestCondition":::
+
+The expression used for `condition` is injected by the compiler into the `message` argument. When a developer calls `Operation` with a `null` argument, the following message is stored in the `ArgumentException`:
```

Comment:
Consider rewording, and in your code InvalidOp not Argument looks like its thrown. 

```suggestion
The expression used for `condition` is injected by the compiler into the `message` argument. When a developer calls `Operation` with a `null` argument, an `InvalidOperationException` is thrown with following message: 
```